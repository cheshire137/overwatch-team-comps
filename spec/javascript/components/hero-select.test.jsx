import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import HeroSelect from '../../../app/assets/javascripts/components/hero-select.jsx'

describe('HeroSelect', () => {
  let component = null
  const hero1 = { id: 1, slug: 'hanzo', image: '/photo.jpg', name: 'Hanzo' }
  let heroIDSelected = hero1.id
  const hero2 = { id: 2, slug: 'ana', image: '/pic.bmp', name: 'Ana' }
  const props = {
    heroes: [hero1, hero2],
    onChange: newHeroID => { heroIDSelected = newHeroID },
    selectedHeroID: heroIDSelected,
    disabled: false,
    isDuplicate: false
  }

  beforeEach(() => {
    component = <HeroSelect {...props} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('can change selected hero', () => {
    const rendered = shallow(component)

    const select = rendered.find('.menu-toggle')
    select.simulate('click', { target: { blur: () => {} } })

    const newHeroButton = rendered.find('.menu-item.hero-ana')
    newHeroButton.simulate('click', {
      preventDefault: () => {},
      target: { blur: () => {} }
    })

    expect(heroIDSelected).toBe(hero2.id)
  })
})
