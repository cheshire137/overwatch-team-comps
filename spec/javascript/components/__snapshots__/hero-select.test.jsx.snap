exports[`HeroSelect matches snapshot 1`] = `
<div
  className="hero-select-container">
  <label
    htmlFor="hero-select-dom-id">
    <img
      alt="Hanzo"
      className="hero-portrait"
      src="/photo.jpg" />
  </label>
  <span
    className="select">
    <select
      disabled={false}
      id="hero-select-dom-id"
      onChange={[Function]}
      value={1}>
      
      <option
        value={1}>
        Hanzo
      </option>
      <option
        value={2}>
        Ana
      </option>
    </select>
  </span>
</div>
`;
